### DISCUSSION TITLE
Make shorter a script

### DISCUSSION THREAD

**Discussion by fzs1933** [Stellarium/stellarium#1585] [Q&A] [answered]
I would like to make a bigger video to show the seasons night sky and wrote a script which looks to good for this purpose.
But it is too long and my programming knowledge is too little to make it simply.
I need help to take this step.

https://github.com/fzs1933/seasons/blob/78422eba0c11eef7cad9a744bd721a93873e03df/nyar_beta.ssc

I emphasize again, my programming knowledge actually approximate to zero.
Thanks a lot for all help/answer.
Zsolt

**Comment by alex-w** [comment, marked as answer]
See "Largest known stars" or "Binocular highlights" scripts for example.

**Reply by fzs1933** [reply]
Thanks Alex!

I don't hope that can interpret it myself.
I'll try it.
Regards

**Reply by fzs1933** [reply]
Hi Alex,

I solved it another way.
But I have one more question. There are two dates and times. For example 2021-03-21T12:00:00 and 2021-03-22T0:0:0. Exactly 12 hours or 720 minutes or 43200 seconds the difference.
I tried the setTimeRate (as far as know the rate is 1, the timer steps 1 seconds. If the rate is 10, the timer steps 10 seconds in 1 seconds; etc...). When I set TimeRate 43200 and core.wait 1, the results is not exactly 43200 seconds. In fact each and every trying bring other and other result.
What is the method to rotate the sky between two date/times?
Thanks
Zsolt

**Reply by fzs1933** [reply]
I can see the core.waitFor, but it is also make different results every trying...

**Reply by fzs1933** [reply]
This:
	core.setDate("2021-03-21T11:00:00", "UTC");
	core.setTimeRate(0);
	core.wait(3);
	core.setTimeRate (7200);
	core.waitFor("2021-03-21T23:00:00", "UTC");
	core.setTimeRate (0);
	core.wait(1);

Always different the results... Why? What am I doing wrong?

**Reply by alex-w** [reply]
Please describe the behaviour of the script, which you want to get (what should be happening between those dates?)

**Reply by fzs1933** [reply]
Thanks your answer!
So, I am looking south. I can see the Sun at 2021-02-21T11:00:00. I want to simulate a half day with exact times.
And I want to rotate the sky to 2021-02-21T23:00:00. Let the rotation be 10 seconds. It is important to the start time and the end time be exactly.
I hope it is clear and enough informations... :)

**Reply by fzs1933** [reply]
![start_to_ends](https://user-images.githubusercontent.com/81897104/113768867-9df8ad80-9720-11eb-8bf5-b87acb9e5785.jpg)
Upper pic is the start
You can see that after two script runs the stop time are not exactly 23:00:00 and not consistent.

**Reply by fzs1933** [reply]
For last.
If the exactly stopping cannot be solved, I will use this method. Could the time equation be confusing? (I switched it off)
Although I thought 43200 seconds is 43200 seconds, regardless of the length of the actual day length.

I make a long video about the nightsky and I would like to export it with the fewest cut. As it possible, there will be only one script and this is why the exact times are very important for me.
